### PR TITLE
Feature/UI 20231226

### LINKED ISSUE #356: 🤖 [MainTask] 機器人聊天室chat-bot-lizzz
流程大綱：

- 規格書撰寫一天
- 功能開發約兩週
  - 串ChatGPT
  - ~use [socket.io](https://socket.io/)~

---

預計子任務（先列出，之後再開單）：

- [x] #357
- [x] 參考網路上的機器人聊天室＆決定參考對象
- [x] 定義使用的所有技術套件等
- [x] #405
- ~查找 socket.io 文件~
- [x] 開發
- [x] open PR
- [ ] PR ok

---

新增子任務：

- [x] #378 
- [x] #379 
- [x] #380 
- [x] #403
- [x] #410
- [x] #414 
- [x] #417
- [x] #411 
- [x] #412
- [x] #423
- [ ] #465
- [ ] #472
- [ ] #476
- [ ] #477

### LINKED ISSUE #423: [Improvement] send icon, text width, delete useless data
- [x] send icon
- [x] text width: 限制寬度使機器人回覆不會與使用者訊息混淆
- [x] delete useless data: 整理不再使用的 dummy data

### PR DESCRIPTION
**Pull request by godmmt**
### DEVELOP
- [x] Homepage
- [x] Message
- [x] Button
- [x] IMessage
- [x] IChatGPTResponse
- [x] getBotResponse

### CHECK LIST
- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 1 (axios)
- new Class / Component: 2
src/components/button.tsx
src/components/message.tsx
- new loop: 1
src/pages/index.tsx
- new recursive: 0
- high risk: 0
- new sql: 0

### UML

- [Class Diagram link](https://lucid.app/lucidchart/7f239953-18e5-41c9-94ed-648e3b3bb53d/edit?viewport_loc=-1651%2C-537%2C1923%2C1267%2C0_0&invitationId=inv_040636e5-7726-4c0c-8b2b-5230594910e8)
- [Sequence Diagram link](https://lucid.app/lucidchart/c918bc56-b89a-4d6f-8493-5037df815983/edit?viewport_loc=2172%2C1298%2C1732%2C1093%2C0_0&invitationId=inv_2ef49e7c-5f81-45de-b53e-667b661043fe)

### CODE REVIEW

File: package-lock.json
Comment (Luphia) [inline package-lock.json:1]: - remove this file
- add package-lock.json to .gitignore

File: tailwind.config.js
Comment (Luphia) [inline tailwind.config.js:1]: - change file type to .ts

File: src/pages/index.tsx [hunk 1]
```diff
@@ -1,31 +1,150 @@
-import { Message } from "@/components/message";
+import Message from '../components/message';
+import Button from '../components/button';
+import { useState } from 'react';
+import { IMessage } from '../interfaces/message';
+import getBotResponse from '../services/chatgpt_service';
+
+const dummyMessages = [
```
Comment (Luphia) [inline src/pages/index.tsx]: - remove dummyMessages
- rename dummyMessages to messages with empty array

File: src/pages/index.tsx [hunk 2]
```diff
@@ -1,31 +1,150 @@
...
+                type="text"
+                value={userInput}
+                placeholder="Say something..."
+                onChange={handleTextInput}
+                onKeyUp={handleKeyUp}
+              />
+            </div>
+            <div className=" text-center	">
+              <Button className="text-amber-500" onClick={handleSendMessage}>
+                Send
```
Comment (Luphia) [inline src/pages/index.tsx]: - change to paper plane icon

File: src/components/message.tsx [hunk 3]
```diff
@@ -1,3 +1,16 @@
...
+import { IMessage } from '../interfaces/message';
+export interface IMessageProps {
+  message: IMessage;
+}
+
+const Message = ({ message }: IMessageProps) => {
+  const position = message.sender === 'user' ? 'self-end' : 'self-start';
+
+  return (
+    <div className={`${position}	my-1 w-fit rounded-lg bg-zinc-200	 px-2 py-1	`}>
```
Comment (Luphia) [inline src/components/message.tsx]: - set max-width to 90% fit (max-w-90)

**Review by Luphia — CHANGES_REQUESTED** [review]

File: src/pages/index.tsx (hunk 1 above)
Comment (godmmt) [reply src/pages/index.tsx]: updated version like this:

截圖 2024-01-10 下午3 32 03

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**